### PR TITLE
Add Beta Dependency Flow

### PR DESCRIPTION
**Pull request by jstvz**
# Critical Changes

# Changes
Adds a flow, `ci_beta_dependencies` to install the latest beta of project dependencies and run tests. Includes task error when running against non-scratch orgs.
# Issues Closed

### CODE REVIEW

File: cumulusci/core/config/BaseProjectConfig.py [hunk 1]
```diff
@@ -402,7 +402,7 @@ def set_org(self, name, org_config):
         self._check_keychain()
         return self.keychain.set_org(name, org_config)
 
-    def get_static_dependencies(self, dependencies=None):
+    def get_static_dependencies(self, dependencies=None, install_latest=None):
```
Comment (cdcarter) [inline cumulusci/core/config/BaseProjectConfig.py]: can you add a note to the docstring indicating what install_latest does?

File: cumulusci/core/config/BaseProjectConfig.py [hunk 2]
```diff
@@ -614,3 +608,20 @@ def process_github_dependency(self, dependency, indent=None):
             repo_dependencies.extend(unpackaged_post)
 
         return repo_dependencies
+
+    def _find_release_version(self, repo, indent, install_latest=None):
```
Comment (cdcarter) [inline cumulusci/core/config/BaseProjectConfig.py]: the arg name is a little confusing I think, install_latest installs the latest pre-release, whereas install_latest=False installs the release at `/latest`...the latest release-release. maybe `include_prereleases`?

File: cumulusci/core/tests/test_config.py [hunk 3]
```diff
@@ -159,13 +159,14 @@ def _get(self, url):
         res.json.return_value = {"name": "2"}
         return res
 
-    def iter_releases(self):
+    def iter_releases(self, count=None):
         return iter(self.releases)
```
[the remark this answers is not in the document]
Comment (davisagli) [reply cumulusci/core/tests/test_config.py]: But wouldn't it do the same thing if you removed the `count` param and made `_find_release_version` call `repo.iter_releases().next()`?

Oh, but the `next` method was renamed to `__next__` in Python 3, so it'd be better to do `next(repo.iter_releases())` for compatibility

File: cumulusci/cumulusci.yml [hunk 4]
```diff
@@ -266,6 +266,20 @@ flows:
                 options:
                     managed: True
 
+    ci_beta_dependencies:
```
[the remark this answers is not in the document]
Comment (jstvz) [reply cumulusci/cumulusci.yml]: `ci_feature_beta_deps` sound any better?

[the remark this answers is not in the document]
Comment (davisagli) [reply cumulusci/cumulusci.yml]: Sure.

**Pull request merged**